python/lsst/daf/butler/direct_query_driver/_driver.py

```diff
@@ -329,7 +346,7 @@ def count(
             builder.postprocessing = Postprocessing()
         if builder.postprocessing:
             if not discard:
-                raise RuntimeError("Cannot count query rows exactly without discarding them.")
+                raise InvalidQueryError("Cannot count query rows exactly without discarding them.")
```

Comment:
This one and the similar one below are the only exceptions that changed type... I think these are roughly in the same spirit as the other InvalidQueryErrors in that the user can fix it by changing one of the parameters in their function call.